master/buildbot/util/identifiers.py

```diff
@@ -17,7 +17,8 @@
 
 from buildbot import util
 
-ident_re = re.compile('^[a-zA-Z\u00a0-\U0010ffff_-][a-zA-Z0-9\u00a0-\U0010ffff_-]*$', flags=re.UNICODE)
+# this line is needed to allow the dot in worker name
+ident_re = re.compile('^[a-zA-Z\u00a0-\U0010ffff._-][a-zA-Z0-9\u00a0-\U0010ffff._-]*$', flags=re.UNICODE)
```

Comment:
Looking at the error message, looks like this one is the correct one.

https://github.com/buildbot/buildbot/blob/d913f1ec10dc7b751d414c090b83644c192d85a4/master/buildbot/config.py#L683

probably we should not depluciate this re, and import the one from util identifier everywhere

Reply:
Yes, this tiny change addresses my problem, as explained in the comment(s).
I added the others just because they seem somehow related. But if some code refactoring is needed, I'm probably not the best candidate to do it.

I'm also surprised by the number of failures this change leads to. I was definitely expecting
```
twisted.trial.unittest.FailTest: no messages for 'a.b.c.d'
```
(which I saw before), but not as many others.

Let me know if I should try to pursue and check unit tests or if there is something more fundamental that you would need to change first (like decreasing the number of regexp copies etc.).

I definitely don't need to change the other two locations for now, and I assume that the number of failed unit tests might decrease after I remove the dot from `identRe = re.compile('^[a-zA-Z._-][a-zA-Z0-9._-]*$')`.